### PR TITLE
Add mastodon and reorganize profile info section

### LINKED ISSUE #12945: Fediverse address on profile page
Hi!
Please add a field for the user's Fediverse address to the profile page. There's only Twitter on this page.

### PR DESCRIPTION
**Pull request by CarlSchwan**
image

image

Fix #12945

### CODE REVIEW

**Comment by blizzz** [comment]
Should this be mastodon in particular, or fediverse or something? As there are other servers that implement the protocol. Pleroma or something, for instance. With the danger sounding to technical. `Fediverse (e.g. Mastodon)` perhaps? `Mastodon/Fediverse`?

**Comment by come-nc** [comment]
Yeah same remark as blizzz, the account I would put in there would actually be Pleroma.
On the other hand fediverse is also a bit generic as that may mean microblog as much as blog or photo or video page.

Maybe we need a generic social media field, with a dropdown on the network used for each value? (and in there we can put any fediverse implementation we want, along with twitter and other stuff)

**Comment by tcitworld** [comment]
This page is starting to be crowded. Maybe @nextcloud/designers might want to have a take at this afterwards?

**Comment by CarlSchwan** [comment]
> Should this be mastodon in particular, or fediverse or something? As there are other servers that implement the protocol. Pleroma or something, for instance. With the danger sounding to technical. `Fediverse (e.g. Mastodon)` perhaps? `Mastodon/Fediverse`?

I changed it to `Fediverse (e.g. Mastodon)`

**Review by Pytal — APPROVED** [review]
Other than the naming of the constants in the code

**Review by jancborchardt — APPROVED** [review]
Nice work @CarlSchwan! :)

**Comment by schiessle** [comment]
Great stuff @CarlSchwan Looking at the screenshots this also fixes https://github.com/nextcloud/server/issues/34641 , right?

File: core/Controller/ClientFlowLoginV2Controller.php [hunk 1]
```diff
@@ -216,7 +216,7 @@ public function apptokenRedirect(string $stateToken, string $user, string $passw
 			return $response;
 		}
 
-		$result = $this->loginFlowV2Service->flowDoneWithAppPassword($loginToken, $this->getServerPath(), $this->userId, $password);
+		$result = $this->loginFlowV2Service->flowDoneWithAppPassword($loginToken, $this->getServerPath(), $user, $password);
```
Comment (nickvergessen) [inline core/Controller/ClientFlowLoginV2Controller.php:219]: This looks totally unrelated?

Comment (CarlSchwan) [reply core/Controller/ClientFlowLoginV2Controller.php:219]: It is. PR to revert is here: https://github.com/nextcloud/server/pull/35556
